### ISSUE TITLE
Servers in a rack do not receive wireless messages sent from each other

### ISSUE THREAD

**Issue by Banderi** [MightyPirates/OpenComputers#3281]
As per title: two servers in the same rack, both with wireless cards, open and listening to a common port - they will receive "modem_message" correctly from outside, but not from each other.

From the Wiki:
> `broadcast(port: number, ...): boolean`
Sends a broadcast message. This message is delivered to all reachable network cards. Returns `true` if the message was sent. **Note that broadcast messages are not delivered to the modem that sent the message.**

Does the "modem" in here actually mean an entire rack acts as one single receiver? This makes little sense to me, since the modem is a component bound to a distinct network/wireless card that you need to put inside the server. I have checked if they by some mistake hooked into each other's card, but that doesn't seem to be the case. `component.modem` returns different addresses.
Maybe a bug carried over from normal wired network card behavior?

The GUI configuration also doesn't seem to matter - I tried various combinations of the settings but I did not notice a difference. It is a bit confusing though, so I could have set them up incorrectly.

EDIT: I tried sending through specific addresses, and still doesn't seem to let the servers receive messages from each other.

**Comment by payonel** [comment]
queue cliche "works for me"

1. make sure the wireless cards, if you cheated them in, have different hardware addresses (they should, we gen addresses for creative-made cards when you place them slots)

2. make sure the port is open on the receiving machine
lua shell:
> component.modem.open(1)

3. run dmesg on the receiving machine
system shell
> dmesg

4. broadcast on port 1
lua shell
> component.modem.broadcast(1, 'hi')

5. profit

**Comment by payonel** [comment]
and to answer your questions

> Does the "modem" in here actually mean an entire rack acts as one single receiver?

no. each network card gets a unique hardware address, and each acts as a unique modem_message origin. each receive and send independently

> Maybe a bug carried over from normal wired network card behavior?

well make sure youre running the latest stuff

> The GUI configuration also doesn't seem to matter - I tried various combinations of the settings but I did not notice a difference. It is a bit confusing though, so I could have set them up incorrectly.

I agree it isn't intuitive but with wireless cards you don't need to attach network sides on the rack

> EDIT: I tried sending through specific addresses, and still doesn't seem to let the servers receive messages from each other.

test with broadcast, less work

**Comment by Banderi** [comment]
I can confirm this does not work, at least when I try it. I was trying with broadcast, but I mentioned the EDIT because I wanted to see if using send() did anything different; `dmesg` (shout out btw, wish I knew about this earlier) reported the broadcast working on other machines, but not the rack's servers - even tried with some encapsulation scripts I made, none work.

Here's my setup, reproducible in a new world:
Servers A and B in a rack, separate C and D machines
If I broadcast/send from C or D all of the other 3 receive and print message on screen; if A or B broadcast, **only** C and D receive the message.
The modems have all different addresses, as I mentioned earlier, and dmesg confirms this;

![image](https://user-images.githubusercontent.com/16541079/80371003-7737c300-8891-11ea-8a64-e0d7baf5a641.png)
![image](https://user-images.githubusercontent.com/16541079/80371692-96832000-8892-11ea-969b-eeb62f5b0bb9.png)

I'm using OC 1.7.4.153 for 1.12.2 (in the latest SF4 modpack), I'm gonna try and update if there's a more up to date version for 1.12, maybe this was a bug that was fixed some time back
